lib/ruby_indexer/test/classes_and_modules_test.rb

```diff
@@ -470,5 +470,51 @@ class ConstantPathReferences
       constant_path_references = T.must(@index["ConstantPathReferences"][0])
       assert_equal(["Foo::Bar", "Foo::Bar2"], constant_path_references.mixin_operation_module_names)
     end
+
+    def test_tracking_singleton_classes
+      index(<<~RUBY)
+        class Foo; end
+        class Foo
+          # Some extra comments
+          class << self
+          end
+        end
+      RUBY
+
+      foo = T.must(@index["Foo::<Class:Foo>"].first)
+      assert_equal(4, foo.location.start_line)
+      assert_equal("Some extra comments", foo.comments.join("\n"))
+    end
+
+    def test_dynamic_singleton_class_blocks
+      index(<<~RUBY)
+        class Foo
+          # Some extra comments
+          class << bar
+          end
+        end
+      RUBY
+
+      singleton = T.must(@index["Foo::<Class:Foo>"].first)
```

Comment:
Maybe this case is should be `Foo::<Class::bar>`? Otherwise, wouldn't its methods get mixed with the actual `Foo::<Class:Foo>` class's?

Reply:
They will indeed. We essentially have two options: mix it with the regular singleton class methods or store it using `Class::bar` (but you wouldn't be able to access those in any way).

I don't mind either approach.

Reply:
I think we should do the latter, i.e. we should mark this as `Foo::<Class:bar>`. However, since we won't be able to track and address single instances, maybe we should have a short-hand for all singleton class of object instances, like `<Class:<instance>>` or something.

Reply:
Done!